src/Resolution.ts

```diff
@@ -486,24 +486,13 @@ export default class Resolution {
   }
 
   /**
-   * Retrieves the domain name from token metadata that is provided by tokenURI from the registry smart contract.
-   * The function will throw an error if the domain in the metadata does not match the hash (e.g. if the metadata is outdated).
+   * Retrieves the domain name from tokenId by parsing registry smart contract event logs.
    * @returns the domain name retrieved from token metadata
    * @param hash - domain hash
    * @param service - nameservice which is used for lookup
    */
   async unhash(hash: string, service: NamingServiceName): Promise<string> {
-    const tokenUri = await this.serviceMap[service].getTokenUri(hash);
-    const metadata = await this.getMetadataFromTokenURI(tokenUri);
-    const receivedHash = this.namehash(metadata.name);
-    if (receivedHash !== hash) {
-      throw new ResolutionError(ResolutionErrorCode.ServiceProviderError, {
```

Comment:
It is hard to believe but it is possible that domain name in `NewURI` event will not match the token id. This may happen if someone would create subdomains in an incorrect way. Like:  `Registry.mintChild(namehash("johnny.crypto"), "www.finance")`. Ideally, `Registry` should not allow `.` inside domain label, but such a check is expensive on chain, so it wasn't implemented.


In such a case a `NewURI` event with `www.financy.johnny.crypto` domain name will be generated, however the namehash of this domain name will not match the on-chain token Id. 

That is why we always need to validate received domain name hash to match the original token id.